### ISSUE TITLE
Support streaming media into existing video/audio tag

### ISSUE THREAD

**Issue by RomanEmelyanov** [feross/webtorrent#531]
As the first step to run WebTorrent in VideoJS player want to run in specific video tag, but Chrome got error: Uncaught TypeError: e.on is not a function

Any idea about file.createReadStream().pipe(video) ?

Full code below:

```
<html><head><script src="https://cdn.jsdelivr.net/webtorrent/latest/webtorrent.min.js"></script></head><body><script>
var client = new WebTorrent()
var torrentId = 'magnet:?xt=urn:btih:6a9759bffd5c0af65319979fb7832189f4f3c35d'
client.add(torrentId, function (torrent) {
var file = torrent.files[0]
var video = document.createElement('video')
video.controls = true
// Chrome error: Uncaught TypeError: e.on is not a function
document.body.appendChild(video)
file.createReadStream().pipe(video)

// Works fine if uncommented
//file.appendTo('body')
})
</script>
</body></html>
```

**Comment by feross** [comment]
Thanks for creating this issue.

The source of the TypeError is that you can't `pipe` a node readable stream into a video tag like that.

We actually used to support `file.createReadStream().pipe(videoTag)` in a previous version of WebTorrent, but I removed it since it was confusing. I was overloading the `ReadableStream.prototype.pipe` semantics and trying to be too magical.

I think the correct solution is to detect when a video or audio tag is passed into `file.appendTo` and do the right thing, i.e. use that existing node. Or perhaps, add a new method `file.streamTo` to be more explicit.

**Comment by RomanEmelyanov** [comment]
Ferros, thanks for fast reply! I'm try to use callback, but got another error:

Uncaught InvalidStateError: Failed to read the 'buffered' property from 'SourceBuffer': This SourceBuffer has been removed from the parent media source.

Any ideas?

Full code:

<html><head>
<link href="http://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet">
<script src="https://cdn.jsdelivr.net/webtorrent/latest/webtorrent.min.js"></script>
</head><body><script>
var client = new WebTorrent()
var torrentId = 'magnet:?xt=urn:btih:6a9759bffd5c0af65319979fb7832189f4f3c35d'
client.add(torrentId, function (torrent) {
var file = torrent.files[0]
file.appendTo('body', function (err, elem) {
 if (err) throw err
 elem.className = 'video-js'
 elem.setAttribute('id','VJSPlayer')
 elem.setAttribute('width','640')
 elem.setAttribute('height','360')
 videojs('VJSPlayer',{},function(){this.load();});
 console.log('New DOM node with the content', elem)
})
})
</script>
<video id="YTAPI" class="video-js" controls preload="none" width="640" height="360" data-setup="{}">
 <source src="http://www.ytapi.com/api/rO0yXC0oyIA/direct/18/" type='video/mp4'>
</video>
 <br>
<script src="http://vjs.zencdn.net/5.4.4/video.js"></script>
</body></html>

**Comment by feross** [comment]
@RomanEmelyanov That's probably caused by using videojs wrong, but I can't help with that because I'm not familiar with videojs. If you find a bug in webtorrent that you can reproduce without videojs, feel free to open a new issue for it.

**Comment by feross** [comment]
To the original issue, there's already a PR out that implements this: https://github.com/feross/webtorrent/pull/425

**Comment by feross** [comment]
Fixed by #425

**Comment by feross** [comment]
Released as 0.65.0.
